lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImplTest.java

```diff
@@ -4214,4 +4222,84 @@ public void testGetNameNotFound() {
 			nodeDao.getNodeName("syn999999");
 		});
 	}
+	
+	@Test
+	public void testFindFirstBoundJsonSchema() throws JSONObjectAdapterException {
+		jsonSchemaTestHelper.truncateAll();
+		// Setup some hierarchy.
+		// grandparent
+		Node grandparent = NodeTestUtils.createNew("grandparent", creatorUserGroupId);
+		grandparent = nodeDao.createNewNode(grandparent);
+		Long grandId = KeyFactory.stringToKey(grandparent.getId());
+		toDelete.add(grandparent.getId());
+		// parent
+		Node parent = NodeTestUtils.createNew("parent", creatorUserGroupId);
+		parent.setParentId(grandparent.getId());
+		parent = nodeDao.createNewNode(parent);
+		Long parentId = KeyFactory.stringToKey(parent.getId());
+		toDelete.add(parent.getId());
+		// child
+		Node child = NodeTestUtils.createNew("child", creatorUserGroupId);
+		child.setParentId(parent.getId());
+		child = nodeDao.createNewNode(child);
+		Long childId = KeyFactory.stringToKey(child.getId());
+		toDelete.add(child.getId());
+
+		String schema$id = "my.org/foo.bar/1.2.3";
+		int index = 0;
+		JsonSchemaVersionInfo schemaInfo = jsonSchemaTestHelper.createNewSchemaVersion(creatorUserGroupId, schema$id,
+				index);
+		// bind the schema to the grand parent
+		JsonSchemaObjectBinding binding = jsonSchemaTestHelper.bindSchemaToObject(creatorUserGroupId, schema$id,
+				grandId, BoundObjectType.entity);
+
+		// call under test
+		Long boundId = nodeDao.findFirstBoundJsonSchema(childId);

```

Comment:
Might want to rename to boundEntityId, also the name of the method findFirstBoundJsonSchema might be findFirstNodeIdBoundJsonSchema